src/commands/kick.cpp

```diff
@@ -0,0 +1,32 @@
+#include "commands.hpp"
+
+static void	kick_user(Message &message, _base_channel &base_channel, Channel &channel, User &author, User &target)
+{
+	if (author.getStatus() == CHANNEL_USER)
+		throw ERR_CHANOPRIVSNEEDED(channel.getName());
+	if (author.getStatus() < target.getStatus())
+		return ;
+	channel.sendMessage(KickMessage(message.peer, target.peer.getNickname(), base_channel, author.peer.getNickname(), message.arguments[2]));
+	channel.remove(target);
+}
+
+int	command_kick(Message &message, Dependencies &deps)
+{
+	if (message.argCount < 2)
+		throw ERR_NEEDMOREPARAMS("KICK");
+
+	if (!_base_channel::isValidName(message.arguments[0]))
+		return (1);
```

Comment:
should throw `InvalidChannelName`

Reply:
what do you think about this suggestion @vfurmane ? Should we forsake the RFC entirely ?

Reply:
`InvalidChannelName` is not an IRC error, thus cannot be thrown without us catching it. Furthermore, we have implemented `_base_channel::isValidName` to avoid big chunks of `trycatch` blocks because of _parse_ methods. Returning 1 is ok, to be right you could throw `ERR_NOSUCHCHANNEL` (which is an IRC error), but I would not require it.